### PR TITLE
[doc] add note about multiple CUDA backends limitation

### PR DESCRIPTION
**Pull request by mkrainiuk**
# Description

As it was mentioned in #270 we have a limitation about building multiple CUDA backends cause problems in functional testing.
This PR adds a note about this limitation to build instructions.

# Checklist

## All Submissions

- [x] Do all unit tests pass locally? Attach a log. N/A
- [x] Have you formatted the code using clang-format? N/A

### CODE REVIEW

**Review by akabalov — APPROVED** [review]

File: docs/building_the_project.rst [hunk 1]
```diff
@@ -585,6 +585,10 @@ CMake.
   the targeted hardware. This backend has only been tested for the ``gfx90a``
   architecture (MI210) at the time of writing.
 
+.. note::
+  Build with multiple CUDA backends enabled could cause testing problems
```
Comment (mmeterel) [inline docs/building_the_project.rst]: I think, the problem is, when functional test flag is enabled and we try to build multiple backends, there will be errors in the build. Can we rephrase the note something like:
`When BUILD_FUNCTIONAL_TESTS is enabled, only single CUDA backend can be built.`

Comment (mkrainiuk) [reply docs/building_the_project.rst]: Good point, let me update the note.

Comment (mkrainiuk) [reply docs/building_the_project.rst]: Updated.

**Review by mmeterel — APPROVED** [review]
Thanks!

**Pull request merged**